sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/PotelSentrySpanProcessor.java

```diff
@@ -69,77 +79,20 @@ public void onStart(final @NotNull Context parentContext, final @NotNull ReadWri
-      if (sampled != null) {
-        // span created by Sentry API
-
-        final @NotNull String traceId = otelSpan.getSpanContext().getTraceId();
-        final @NotNull String spanId = otelSpan.getSpanContext().getSpanId();
-        // TODO [POTEL] parent span id could be invalid
-        final @NotNull String parentSpanId = otelSpan.getParentSpanContext().getSpanId();
-
-        final @NotNull PropagationContext propagationContext =
-            new PropagationContext(
-                new SentryId(traceId),
-                new SpanId(spanId),
-                new SpanId(parentSpanId),
-                baggage,
-                sampled);
-
-        scopes.configureScope(
-            scope -> {
-              scope.withPropagationContext(
-                  oldPropagationContext -> {
-                    scope.setPropagationContext(propagationContext);
-                  });
-            });
-
-        // TODO [POTEL] can we use OTel Sampler to let OTel know our sampling decision
-        // Sentry not sampled vs OTel not sampled may mean different things for trace propagation
-        samplingDecision =
-            new TracesSamplingDecision(
-                sampled,
-                sampleRate,
-                profileSampled == null ? false : profileSampled,
-                profileSampleRate);
-      } else {
-        // span not created by Sentry API
-
-        final @NotNull String traceId = otelSpan.getSpanContext().getTraceId();
-        final @NotNull String spanId = otelSpan.getSpanContext().getSpanId();
-        final @NotNull SpanId sentrySpanId = new SpanId(spanId);
-
-        @Nullable
-        SentryTraceHeader sentryTraceHeader = parentContext.get(SentryOtelKeys.SENTRY_TRACE_KEY);
-        @Nullable Baggage baggageFromContext = parentContext.get(SentryOtelKeys.SENTRY_BAGGAGE_KEY);
-        if (sentryTraceHeader != null) {
-          baggage = baggageFromContext;
-        }
 
-        final @NotNull PropagationContext propagationContext =
-            sentryTraceHeader == null
-                ? new PropagationContext(new SentryId(traceId), sentrySpanId, null, baggage, null)
-                : PropagationContext.fromHeaders(sentryTraceHeader, baggage, sentrySpanId);
-
-        scopes.configureScope(
-            scope -> {
-              scope.withPropagationContext(
-                  oldPropagationContext -> {
-                    scope.setPropagationContext(propagationContext);
-                  });
-            });
-
-        final @NotNull TransactionContext transactionContext =
-            TransactionContext.fromPropagationContext(propagationContext);
-        samplingDecision = tracesSampler.sample(new SamplingContext(transactionContext, null));
-      }
+      // TODO [POTEL] what do we use as fallback here? could happen if misconfigured (i.e. sampler
+      // not in place)
+      final boolean sampled = samplingDecision != null ? samplingDecision.getSampled() : true;
```

Comment:
Hmm, should we use `otelSpan.getSpanContext().isSampled()` as fallback?

Reply:
Hmm not sure